src-ghc/Pact/ApiReq.hs

```diff
@@ -114,6 +127,52 @@ mkExec code mdata addy kps ridm = do
     (pack $ show rid)
     (Exec (ExecMsg (pack code) mdata))
 
+mkApiReqCont :: ApiReq -> FilePath -> IO ((ApiReq,String,Value,Maybe Address),Command Text)
+mkApiReqCont ar@ApiReq{..} fp = do
+  txId <- case _ylTxId of
+    Just t  -> return t
+    Nothing -> dieAR "Expected a 'txid' entry"
+    
+  step <- case _ylStep of
+    Just s  -> return s
+    Nothing -> dieAR "Expected a 'step' entry"
+    
+  rollback <- case _ylRollback of
+    Just r  -> return r
+    Nothing -> dieAR "Expected a 'rollback' entry"
+
+  oldCwd <- getCurrentDirectory
```

Comment:
refactor 144-146 to a "withCurrentDirectory" function, and use with `mkApiReqExec` too

Reply:
that name isn't good ... maybe `withDirectoryChange`